_doc-build-windows/action.yml

```diff
@@ -212,47 +212,95 @@ runs:
+      shell: powershell
+      run: |
+        if ((Test-Path "pyproject.toml") -and (Get-Content "pyproject.toml" | Select-String -Pattern 'build-backend = "poetry\.core\.masonry\.api"')) {
+          echo "BUILD_BACKEND=$(echo 'poetry')" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "SPHINX_BUILD_MAKE=$(echo 'poetry run -- make.bat')" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        } else {
+          echo "BUILD_BACKEND=$(echo 'pip')" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "SPHINX_BUILD_MAKE=$(echo '.\make.bat')" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        }
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Build backend: ${{ env.BUILD_BACKEND }}
+          Sphinx build make: ${{ env.SPHINX_BUILD_MAKE }}
+
+    # ------------------------------------------------------------------------
+
     - uses: ansys/actions/_logging@main
       with:
         level: "INFO"
         message: >
           Set up python to build the documentation.
 
-    - name: "Update pip"
+    - name: "Set poetry environment variable(s) (if required)"
       shell: powershell
-      run: python -m pip install -U pip
+      if: env.BUILD_BACKEND == 'poetry'
+      run: |
+        # For projects using poetry, do not use a virtual environment.
+        # Poetry uses virtual environments to install its dependencies, but this might
+        # lead to problems if it is not activated prior to executing poetry commands.
+        # Store POETRY_VIRTUALENVS_CREATE=false in the GitHub environment to prevent
+        # poetry from creating a virtual environment.
+        #
+        echo "POETRY_VIRTUALENVS_CREATE=false" >> $GITHUB_ENV
 
-    - name: Check if requirements.txt file exists
+    - name: "Update pip"
       shell: powershell
       run: |
-        echo "EXISTS_DOC_REQUIREMENTS=$(if (Test-Path '${{ inputs.requirements-file }}') { echo 'true' } else { echo 'false' })" | Out-File -Append -FilePath $env:GITHUB_ENV -Encoding utf8
+        .venv\Scripts\Activate.ps1
+        python -m pip install -U pip
 
-    - name: Print previous output
+    - name: "Check if requirements.txt file exists"
       shell: powershell
       run: |
-        echo "Output was found ${{ env.EXISTS_DOC_REQUIREMENTS }}"
+        echo "EXISTS_DOC_REQUIREMENTS=$(if (Test-Path '${{ inputs.requirements-file }}') { echo 'true' } else { echo 'false' })" | Out-File -Append -FilePath $env:GITHUB_ENV -Encoding utf8
 
     - name: Install documentation dependencies from requirements file
       shell: powershell
       if: ${{ env.EXISTS_DOC_REQUIREMENTS == 'true' }}
       run: |
+        .venv\Scripts\Activate.ps1
         python -m pip install -r ${{ inputs.requirements-file }}
 
+    - name: "Install poetry (if required)"
+      shell: powershell
+      if: env.BUILD_BACKEND == 'poetry'
+      run: |
+        .venv\Scripts\Activate.ps1
```

Comment:
Same here. I won't call out every location poetry is installed, but it should always be installed in a separate environment. pipx and runner.temp is a convenient way of doing that, especially on persistent infrastructure.